fenix/app/src/test/java/org/mozilla/fenix/shopping/ReviewQualityCheckFeatureTest.kt

```diff
@@ -394,4 +418,138 @@ class ReviewQualityCheckFeatureTest {
         tested.start()
         assertFalse(isExpanded!!)
     }
+
+    @Test
+    fun `GIVEN feature is enabled WHEN non product url accessed THEN callback not called`() {
+        runTest {
+            var invokedCounter = 0
+            val tab = createTab(
+                url = "https://www.mozilla.org",
+                id = "test-tab",
+                isProductUrl = false,
+            )
+            val browserState = BrowserState(
+                tabs = listOf(tab),
+                selectedTabId = tab.id,
+            )
+            val tested = ReviewQualityCheckFeature(
+                appStore = AppStore(),
+                browserStore = BrowserStore(
+                    initialState = browserState,
+                ),
+                shoppingExperienceFeature = FakeShoppingExperienceFeature(),
+                onIconVisibilityChange = {},
+                onBottomSheetStateChange = {},
```

Comment:
Let's also add `debounceTimeoutMillis = { 0 },` to make sure the callback is not being called because of `isProductUrl` and not any other reason.